### ISSUE TITLE
RecursionError via _merge_name_docs

### ISSUE THREAD

**Issue by blueyed** [davidhalter/jedi#1490]
https://github.com/davidhalter/jedi/commit/e3e6727a2d1b037d73f25462b9b080afad24b25a appears to trigger a RecursionError with jedi-vim's tests, after triggering completion with `from os import `.

<details>
<summary>
RecursionError: maximum recursion depth exceeded while calling a Python object
</summary>

```
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 235, in py__doc__
    return _merge_name_docs(names)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 22, in _merge_name_docs
    doc += name.py__doc__()
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 633, in py__doc__
    return _merge_name_docs(names)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 22, in _merge_name_docs
    doc += name.py__doc__()
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 235, in py__doc__
    return _merge_name_docs(names)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 22, in _merge_name_docs
    doc += name.py__doc__()
  …
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 22, in _merge_name_docs
    doc += name.py__doc__()
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/names.py", line 627, in py__doc__
    names = convert_names(names, prefer_stub_to_compiled=False)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/conversion.py", line 140, in convert_names
    return _try_stub_to_python_names(
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/utils.py", line 20, in wrapper
    return list(func(*args, **kwargs))
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/conversion.py", line 63, in _try_stub_to_python_names
    values = convert_values(name.infer(), ignore_compiled=prefer_stub_to_compiled)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/conversion.py", line 154, in convert_values
    return ValueSet.from_sets(
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/common/value.py", line 32, in from_sets
    for set_ in sets:
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/conversion.py", line 155, in <genexpr>
    _stub_to_python_value_set(stub_value, ignore_compiled=ignore_compiled)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/conversion.py", line 28, in _stub_to_python_value_set
    values = _infer_from_stub(stub_module_context, qualified_names, ignore_compiled)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/conversion.py", line 50, in _infer_from_stub
    non_stubs = non_stubs.py__getattribute__(name)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/base_value.py", line 395, in py__getattribute__
    return ValueSet.from_sets(c.py__getattribute__(*args, **kwargs) for c in self._set)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/common/value.py", line 32, in from_sets
    for set_ in sets:
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/base_value.py", line 395, in <genexpr>
    return ValueSet.from_sets(c.py__getattribute__(*args, **kwargs) for c in self._set)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/base_value.py", line 88, in py__getattribute__
    names = self.goto(name_or_str, name_context, analysis_errors)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/base_value.py", line 77, in goto
    names = finder.filter_name(filters, name_or_str)
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/finder.py", line 35, in filter_name
    for filter in filters:
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/base_value.py", line 63, in _get_value_filters
    for f in self.get_filters(origin_scope=origin_scope):
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/value/module.py", line 98, in get_filters
    for star_filter in self.iter_star_filters():
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/value/module.py", line 141, in iter_star_filters
    yield next(star_module.get_filters())
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/gradual/stub_value.py", line 44, in get_filters
    next(filters)  # Ignore the first filter and replace it with our own
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/value/module.py", line 90, in get_filters
    ParserTreeFilter(
  File "…/vim/plugged/jedi-vim/pythonx/jedi/jedi/inference/filters.py", line 114, in __init__
    super(ParserTreeFilter, self).__init__(parent_context, node_context.tree_node)
RecursionError: maximum recursion depth exceeded while calling a Python object
```

</details>

**Comment by davidhalter** [comment]
I'll look into it.

**Comment by davidhalter** [comment]
Fixed in eee919174d123065965bebc6fb8f1ecb2b683a8c

We should be find just taking the master branch for Jedi in jedi-vim this time.

I will probably create a release in a few weeks, but we don't have to wait IMO.
